### PR TITLE
feat: [M3 7015] - Create nodebalancer tab in firewalls landing

### PR DESCRIPTION
**Pull request by tyler-akamai**
## Description 📝
Added NodeBalancers in the Firewalls landing page.

## Major Changes 🔄
- Added new NodeBalancer tab
- Added the total number of each product as an integer next to the tab label

## Preview 📷
| Before  | After   |
| ------- | ------- |
| Screenshot 2023-08-24 at 1 48 55 PM | Screenshot 2023-08-24 at 1 49 06 PM |
|  | Screenshot 2023-08-24 at 1 50 10 PM |

## How to test 🧪
1. Navigate to http://localhost:3000/firewalls
2. Click on any of your firewalls
3. Ensure there is an extra NodeBalancers tab
4. Ensure the number next to the Linodes tab accurately represents the number of assigned linodes
5. Ensure the Linode tab functionality still works
6. Ensure the breadcrumb now has "Assigned [Device]", device is either a Linode or NodeBalancer

### CODE REVIEW

File: packages/manager/src/features/Firewalls/FirewallDetail/index.tsx [hunk 1]
```diff
@@ -112,15 +149,24 @@ export const FirewallDetail = () => {
           <SafeTabPanel index={0}>
             <FirewallRulesLanding
               disabled={!userCanModifyFirewall}
-              firewallID={firewallId}
+              firewallID={firewallID}
               rules={firewall.rules}
             />
           </SafeTabPanel>
           <SafeTabPanel index={1}>
-            <FirewallLinodesLanding
```
Comment (tyler-akamai) [inline packages/manager/src/features/Firewalls/FirewallDetail/index.tsx:120]: There would be a lot of commonality between the FirewallsLinodeLanding and the FirewallsNodeBalancerLanding, so I decided to make a generic FirewallsDeviceLanding that contains either a table of Linodes or a table of NodeBalancers. If we want more customization, we might have to look into having two separate files.

**Review by jcallahan-akamai — CHANGES_REQUESTED** [review]
Great work @TylerWJ. 

We need to fix the breadcrumb. The "Assigned Linodes" part should not be editable. Let's check with UX to see what the options are here.

Everything else LGTM.

Screenshot 2023-08-24 at 4 13 54 PM

**Comment by carrillo-erik** [comment]
Just my small change request, everything looks good to me. My other suggestion would be to add some unit tests.

**Review by jcallahan-akamai — APPROVED** [review]

**Review by bnussman-akamai — COMMENTED** [review]
Should we feature flag this?

**Review by carrillo-erik — APPROVED** [review]

**Comment by tyler-akamai** [comment]
> Should we feature flag this?

I created a feature branch for all of our changes and then when we are done we will merge it into develop.

**Pull request merged**